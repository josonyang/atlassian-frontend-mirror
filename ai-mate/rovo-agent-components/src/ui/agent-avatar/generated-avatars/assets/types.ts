export type AvatarIconProps = {
	size: number;
	primaryColor: string;
	secondaryColor: string;
};
