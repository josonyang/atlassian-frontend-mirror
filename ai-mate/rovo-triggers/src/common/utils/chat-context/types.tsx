export type ChatContextState = {
	[contextKey: string]: unknown;
};
export type ChatContextPayload = {
	contextKey: string;
	context: <T>(chatContext: T) => T;
};
