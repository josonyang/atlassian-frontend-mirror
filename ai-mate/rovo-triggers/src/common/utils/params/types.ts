export type RovoChatPathway = 'chat' | 'agents-browse' | 'agents-create';

export interface BaseRovoChatParams {
	pathway: RovoChatPathway;
	agentId: string;
	conversationId: string;
	prompt: string;
	cloudId: string;
	triggerOpen: boolean;
}

export type ValidPrefix = 'rovoChat';
export type ValidParam = keyof BaseRovoChatParams;
export type ValidPrefixedParam = `${ValidPrefix}${ValidParam}`;

type RovoParams<T extends RovoChatPathway, P = object> = BaseRovoChatParams & {
	pathway: T;
} & P;

type ChatParams = RovoParams<'chat'>;
type AgentBrowseParams = RovoParams<'agents-browse'>;
type AgentCreateParams = RovoParams<'agents-create'>;

export type RovoChatParams = Partial<ChatParams | AgentCreateParams | AgentBrowseParams>;
