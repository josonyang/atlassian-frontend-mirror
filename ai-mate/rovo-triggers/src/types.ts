import type { DocNode } from '@atlaskit/adf-schema';

import type { ChatContextPayload } from './common/utils/chat-context';

export const Topics = {
	AI_MATE: 'ai-mate',
} as const;
export type Topic = (typeof Topics)[keyof typeof Topics];

export type PayloadCore<TKey extends string, TData = void> = {
	type: TKey;
	source: string;
	openChat?: boolean;
	product?: string;
	interactionSource?: string;
} & (TData extends void ? {} : { data: TData });

export type MessageSendPayload = PayloadCore<
	'message-send',
	{
		prompt: string;
		productKey?: string;
	}
>;

export type ChatClosePayload = PayloadCore<'chat-close', {}>;

// Can only specify either `agentId` or `agentExternalConfigReference`, not both
type TargetAgentParam =
	| {
			agentId: string;
			agentExternalConfigReference?: never;
	  }
	| {
			agentId?: never;
			agentExternalConfigReference: string;
	  };

export type ChatNewPayload = PayloadCore<
	'chat-new',
	{
		name?: string;
		// Used to prefab the chat history
		dialogues: Array<{
			human_message: {
				content: string;
				mimeType?: 'text/markdown' | 'text/adf';
			};
			agent_message: {
				content: string;
				mimeType?: 'text/markdown' | 'text/adf';
			};
		}>;
		// Used for follow-up prompt once chat is created
		prompt?: string | DocNode;
		files?: UploadedFile[];
		contentContext?: 'staging-area' | 'global';
		sourceId?: string;
		minionAlias?: string;
	} & Partial<TargetAgentParam>
>;

export type EditorContextPayloadData =
	| {
			document: {
				type: 'text/markdown' | 'text/adf';
				content: string;
			};
			selection: {
				type: 'text/markdown' | 'text/plain';
				content: string;
			};
			selectionFragment?: string;
			selectionLocalIds?: string;
			isViewMode?: boolean;
	  }
	| undefined;

export type WhiteboardContextPayloadData =
	| {
			type: 'image/svg+xml' | 'text/plain';
			content: string;
			contentId?: string;
			isViewMode?: boolean;
	  }
	| undefined;

export type DatabaseContextPayloadData =
	| {
			contentId: string;
			csv: string;
			title: string;
			url: string;
	  }
	| undefined;

export type BrowserContextPayloadData = {
	context:
		| {
				browserUrl: string;
				htmlBody?: string;
				canvasText?: string;
		  }
		| undefined;
};

export type WorkflowContextPayloadData = {
	currentWorkflowDocument?: Record<string, unknown>;
};

// Not using the PayloadCore because the `data: type | undefined` is necessary
// but `| undefined` will cause `data` to be removed by PayloadCore
export type EditorContextPayload = PayloadCore<'editor-context-payload'> & {
	data: EditorContextPayloadData;
};

// Not using the PayloadCore because the `data: type | undefined` is necessary
// but `| undefined` will cause `data` to be removed by PayloadCore
export type BrowserContextPayload = PayloadCore<'browser-context-payload'> & {
	data: BrowserContextPayloadData;
};

// Not using the PayloadCore because the `data: type | undefined` is necessary
// but `| undefined` will cause `data` to be removed by PayloadCore
export type WhiteboardContextPayload = PayloadCore<'whiteboard-context-payload'> & {
	data: WhiteboardContextPayloadData;
};

export type ChatDraftPayload = PayloadCore<'chat-draft'>;

export type OpenBrowseAgentPayload = PayloadCore<'open-browse-agent-modal'>;

export type OpenBrowseAgentSidebarPayload = PayloadCore<'open-browse-agent-sidebar'>;

export type EditorSuggestionPayload = PayloadCore<
	'editor-suggestion',
	{
		mode: 'insert' | 'replace';
		content: string;
		agentId?: string;
	}
>;

export type EditorAgentChangedPayload = PayloadCore<
	'agent-changed',
	{
		agent: {
			id: string;
			name: string;
			identityAccountId?: string | null;
		} | null;
	}
>;

export type ChatOpenPayload = PayloadCore<
	'chat-open',
	{
		channelId: string;
		// @deprecated this is not being used, please use `chat-new` if you want to open a new chat with an agent
		agentId?: string;
	}
>;

export type ForgeAppAuthSuccess = PayloadCore<
	'forge-auth-success',
	{
		is3pActionAuth?: boolean;
	}
>;
export type ForgeAppAuthFailure = PayloadCore<
	'forge-auth-failure',
	{
		is3pActionAuth?: boolean;
		errorMessage: string | undefined;
	}
>;

/** Inserts a prompt into the chat input - either:
 * - A prompt without a placeholder - sends as a message
 * - A prompt with a placeholder - inserts the prompt into the chat input
 */
export type InsertPromptPayload = PayloadCore<
	'insert-prompt',
	{
		prompt: string;
		/**
		 * Overrides the default placeholder type
		 */
		placeholderType?: 'person' | 'link' | 'generic';
	}
>;

export type TransitionId = string;
export type StatusId = string;
export type StatusCategory = 'TODO' | 'IN_PROGRESS' | 'DONE' | 'UNDEFINED';
export type RuleConfig = {
	[key: string]: string;
};
export type AddStatusRovoPayload = {
	statusId: StatusId;
	statusName: string;
	statusCategory: StatusCategory;
};
export type UpdateStatusRovoPayload = {
	oldStatusName: string;
	oldStatusCategory: StatusCategory;
	newStatusName: string;
	newStatusCategory: StatusCategory;
};
export type DeleteStatusRovoPayload = {
	statusId: string;
	statusName: string;
	statusCategory: StatusCategory;
};
export type AddNewTransitionRovoPayload = {
	id: TransitionId;
	name: string;
	toStatusId: StatusId;
	toStatusName: string;
	toStatusCategory: StatusCategory;
	links: {
		fromStatusId: StatusId;
		fromStatusName: string;
		fromStatusCategory: StatusCategory;
	}[];
};
export type UpdateTransitionRovoPayload = {
	id: TransitionId;
	name: string;
	toStatusId: StatusId;
	toStatusName: string;
	toStatusCategory: StatusCategory;
	links: {
		fromStatusId: StatusId;
		fromStatusName: string;
		fromStatusCategory: StatusCategory;
	}[];
};
export type DeleteTransitionRovoPayload = {
	transitionId: TransitionId;
	transitionName: string;
};
export type AddRuleRovoPayload = {
	ruleTemplateKey: string;
	ruleConfig: RuleConfig;
	ruleDescription: string;
	transitionId: TransitionId;
};
export type UpdateRuleRovoPayload = {
	ruleTemplateKey: string;
	ruleConfig: RuleConfig;
	transitionId: TransitionId;
	ruleIdBeingEdited: string;
	ruleDescription: string;
};
export type DeleteRuleRovoPayload = {
	ruleId: string;
	ruleTemplateKey: string;
	ruleDescription: string;
	transitionId: TransitionId;
};

export type JiraWorkflowWizardAction =
	| { operationType: 'ADD_STATUS'; payload: AddStatusRovoPayload }
	| { operationType: 'UPDATE_STATUS'; payload: UpdateStatusRovoPayload }
	| { operationType: 'DELETE_STATUS'; payload: DeleteStatusRovoPayload }
	| { operationType: 'ADD_TRANSITION'; payload: AddNewTransitionRovoPayload }
	| { operationType: 'UPDATE_TRANSITION'; payload: UpdateTransitionRovoPayload }
	| { operationType: 'DELETE_TRANSITION'; payload: DeleteTransitionRovoPayload }
	| { operationType: 'ADD_RULE'; payload: AddRuleRovoPayload }
	| { operationType: 'UPDATE_RULE'; payload: UpdateRuleRovoPayload }
	| { operationType: 'DELETE_RULE'; payload: DeleteRuleRovoPayload };

export type JiraWorkflowWizardActionsPayload = PayloadCore<
	'jira-workflow-wizard-actions',
	{
		operations?: JiraWorkflowWizardAction[];
		currentWorkflowDocument?: Record<string, unknown>;
	}
>;

export type DashboardInsightsActionsPayload = PayloadCore<'dashboard-insights-actions'> & {
	data?: DashboardInsightsActionsPayloadData;
};

export type DashboardInsightsActionsPayloadData =
	| {
			content: string;
	  }
	| undefined;

export type SetChatContextPayload = PayloadCore<'set-message-context', ChatContextPayload>;

export type Payload =
	| MessageSendPayload
	| ChatClosePayload
	| ChatNewPayload
	| ChatDraftPayload
	| EditorContextPayload
	| ChatOpenPayload
	| OpenBrowseAgentPayload
	| OpenBrowseAgentSidebarPayload
	| EditorSuggestionPayload
	| EditorAgentChangedPayload
	| BrowserContextPayload
	| WhiteboardContextPayload
	| ForgeAppAuthSuccess
	| ForgeAppAuthFailure
	| JiraWorkflowWizardActionsPayload
	| InsertPromptPayload
	| DashboardInsightsActionsPayload
	| SetChatContextPayload;

export type Callback = (payload: Payload) => void;

export type TopicEvents = {
	[key in Topic]?: Array<{
		id: string;
		callback: Callback;
	}>;
};

export type TopicEventQueue = {
	[key in Topic]?: Payload;
};

export type UploadedFile = {
	id: string;
	conversation_channel_id?: string;
	blob_id?: string;
	name: string;
	size?: number;
	type?: string;
	created_at?: string;
	owner?: string;
	isLoading: boolean;
	error?: string;
	fileObject?: File;
};
