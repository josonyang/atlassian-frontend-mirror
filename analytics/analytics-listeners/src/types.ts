import type Logger from './helpers/logger';
import { type GasPurePayload, type GasPureScreenEventPayload } from '@atlaskit/analytics-gas-types';

export type AnalyticsWebClient = {
	sendOperationalEvent: (event: GasPurePayload) => void;
	sendScreenEvent: (event: GasPureScreenEventPayload) => void;
	sendTrackEvent: (event: GasPurePayload) => void;
	sendUIEvent: (event: GasPurePayload) => void;
};

export type ListenerProps = {
	children?: React.ReactNode;
	client?: AnalyticsWebClient | Promise<AnalyticsWebClient>;
	logger: Logger;
};

export enum FabricChannel {
	atlaskit = 'atlaskit',
	elements = 'fabric-elements',
	navigation = 'navigation',
	editor = 'editor',
	media = 'media',
	peopleTeams = 'peopleTeams',
	notifications = 'notifications',
	recentWork = 'recentWork',
	atlas = 'atlas',
	crossFlow = 'crossFlow',
	linkingPlatform = 'linkingPlatform',
	postOffice = 'postOffice',
	aiMate = 'aiMate',
	avp = 'avp',
	growth = 'growth',
	omniChannel = 'omniChannel',
}
