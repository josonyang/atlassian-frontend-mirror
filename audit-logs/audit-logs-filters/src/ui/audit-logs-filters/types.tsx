export interface AuditLogsFiltersProps {
	/**
	 * When `true` AuditLogsFilters will appear selected.
	 */
	isSelected?: boolean;

	/**
	 * A `testId` prop is provided for specified elements, which is a unique
	 * string that appears as a data attribute `data-testid` in the rendered code,
	 * serving as a hook for automated tests */
	testId?: string;

	/**
	 * Width of AuditLogsFilters.
	 */
	width?: number;
}

export type ProgressBarProps = { value: number };
