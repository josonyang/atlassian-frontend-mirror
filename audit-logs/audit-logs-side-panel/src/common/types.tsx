export interface AuditLogEventData {
	id: string;
	type: string;
	message: string;
	attributes: {
		action: string;
		actor: {
			id: string | null;
			name: string | null;
			email: string | null;
			picture: string | null;
			links: {
				alt: string | null;
				self: string | null;
			} | null;
		} | null;
		location: {
			ip: string | null;
			city: string | null;
		} | null;
		time: string;
	};
}

export interface AuditLogsSidePanelProps {
	/**
	 * The audit log event data to display
	 */
	event: AuditLogEventData | undefined;
}
