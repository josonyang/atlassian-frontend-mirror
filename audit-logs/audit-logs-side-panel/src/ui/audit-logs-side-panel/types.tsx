export interface AuditLogsSidePanelProps {
	/**
	 * When `true` AuditLogsSidePanel will appear selected.
	 */
	isSelected?: boolean;

	/**
	 * A `testId` prop is provided for specified elements, which is a unique
	 * string that appears as a data attribute `data-testid` in the rendered code,
	 * serving as a hook for automated tests */
	testId?: string;

	/**
	 * Width of AuditLogsSidePanel.
	 */
	width?: number;
}

export type ProgressBarProps = { value: number };
