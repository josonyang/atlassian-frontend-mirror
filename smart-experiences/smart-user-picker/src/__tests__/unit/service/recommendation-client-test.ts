import fetchMock from 'fetch-mock/cjs/client';

import getUserRecommendations from '../../../service/recommendation-client';
import { type RecommendationRequest } from '../../../types';
import { type IntlShape } from 'react-intl-next';

const URS_URL = '/gateway/api/v1/recommendations';

const intl = {} as IntlShape;

const exampleContext = {
  childObjectId: 'childObjectId',
  containerId: 'containerId',
  contextType: 'fieldId',
  objectId: 'objectId',
  principalId: 'principalId',
  productAttributes: {
    emailDomain: 'Context',
    isPublicRepo: false,
    workspaceIds: ['02b941e3-cfaa-40f9-9a58-cec53e20bdc3'],
    pullRequestId: 18056,
  },
  productKey: 'bitbucket',
  sessionId: 'session-id',
  siteId: 'site-id',
};

const exampleRequest: RecommendationRequest = {
  context: exampleContext,
  maxNumberOfResults: 50,
  query: 'query',
  includeUsers: true,
  includeGroups: true,
  includeTeams: true,
  includeNonLicensedUsers: false,
};

const exampleResponse = {
  status: 200,
  body: JSON.stringify({
    recommendedUsers: [
      {
        entityType: 'USER',
        id: '5ee0adc79583380ab0afec31',
        name: 'John Smith',
        email: 'jsmith@atlassian.com',
        avatarUrl:
          'https://avatar-management--avatars.us-west-2.prod.public.atl-paas.net',
        nickname: 'John Smith',
        matchPositions: {},
        accessLevel: 'CONTAINER',
        accountStatus: 'ACTIVE',
        notMentionable: false,
        locale: 'en-GB',
        title: 'Development Manager',
        userType: 'DEFAULT',
      },
      {
        entityType: 'USER',
        id: '5ee0adc79583380ab0afec32',
        name: 'John Doe',
        email: 'jsmith@atlassian.com',
        avatarUrl:
          'https://avatar-management--avatars.us-west-2.prod.public.atl-paas.net',
        nickname: 'John Doe',
        matchPositions: {},
        accessLevel: 'CONTAINER',
        accountStatus: 'ACTIVE',
        notMentionable: false,
        locale: 'en-GB',
        title: 'Development Manager',
        userType: 'DEFAULT',
      },
    ],
  }),
};

describe('default-value-hydration-client', () => {
  afterEach(() => {
    jest.clearAllMocks();
    fetchMock.restore();
  });

  it('should reject promise on network error', async () => {
    fetchMock.post(URS_URL, 504);

    try {
      await getUserRecommendations(exampleRequest, intl);
    } catch (error) {
      expect((error as any).message).toMatchSnapshot('URS error');
    }
  });

  it('should translate given context to correct fetch request', async () => {
    let requestBody;

    fetchMock.post(
      {
        functionMatcher: (url: string, options: any) => {
          requestBody = JSON.parse(options.body);

          return url === '/gateway/api/v1/recommendations';
        },
      },
      exampleResponse,
      {
        repeat: 1,
        overwriteRoutes: false,
      },
    );

    const users = await getUserRecommendations(exampleRequest, intl);

    expect(fetchMock.called()).toBeTruthy();
    expect(requestBody).toMatchSnapshot('URS query');
    expect(users).toMatchSnapshot('URS users');
  });
});
