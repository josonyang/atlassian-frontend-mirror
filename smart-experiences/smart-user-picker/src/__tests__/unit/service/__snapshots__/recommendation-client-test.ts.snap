// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default-value-hydration-client should reject promise on network error: URS error 1`] = `"error calling smart service, statusCode=504, statusText=Gateway Timeout"`;

exports[`default-value-hydration-client should translate given context to correct fetch request: URS query 1`] = `
Object {
  "context": Object {
    "childObjectId": "childObjectId",
    "containerId": "containerId",
    "contextType": "fieldId",
    "objectId": "objectId",
    "principalId": "principalId",
    "productAttributes": Object {
      "emailDomain": "Context",
      "isPublicRepo": false,
      "pullRequestId": 18056,
      "workspaceIds": Array [
        "02b941e3-cfaa-40f9-9a58-cec53e20bdc3",
      ],
    },
    "productKey": "bitbucket",
    "sessionId": "session-id",
    "siteId": "site-id",
  },
  "includeGroups": true,
  "includeNonLicensedUsers": false,
  "includeTeams": true,
  "includeUsers": true,
  "maxNumberOfResults": 50,
  "performSearchQueryOnly": false,
  "searchQuery": Object {
    "cpusQueryHighlights": Object {
      "field": "",
      "query": "",
    },
    "customQuery": "",
    "customerDirectoryId": "",
    "filter": "",
    "minimumAccessLevel": "APPLICATION",
    "queryString": "query",
    "restrictTo": Object {
      "groupIds": Array [],
      "userIds": Array [],
    },
    "searchUserbase": false,
  },
}
`;

exports[`default-value-hydration-client should translate given context to correct fetch request: URS users 1`] = `
Array [
  Object {
    "avatarUrl": "https://avatar-management--avatars.us-west-2.prod.public.atl-paas.net",
    "email": "jsmith@atlassian.com",
    "id": "5ee0adc79583380ab0afec31",
    "isExternal": false,
    "lozenge": undefined,
    "name": "John Smith",
    "sources": undefined,
    "title": "Development Manager",
    "tooltip": "John Smith",
    "type": "user",
  },
  Object {
    "avatarUrl": "https://avatar-management--avatars.us-west-2.prod.public.atl-paas.net",
    "email": "jsmith@atlassian.com",
    "id": "5ee0adc79583380ab0afec32",
    "isExternal": false,
    "lozenge": undefined,
    "name": "John Doe",
    "sources": undefined,
    "title": "Development Manager",
    "tooltip": "John Doe",
    "type": "user",
  },
]
`;
